commit c04cf9e14f109ebcc425c1efd2c01294c52a4d62
Author: Mario Limonciello <mario.limonciello@amd.com>
Date:   Fri Jun 23 08:49:55 2023 -0500

    crypto: ccp - Add support for fetching a nonce for dynamic boost control
    
    Dynamic Boost Control is a feature offered on AMD client platforms that
    allows software to request and set power or frequency limits.
    
    Only software that has authenticated with the PSP can retrieve or set
    these limits.
    
    Create a character device and ioctl for fetching the nonce. This ioctl
    supports optionally passing authentication information which will influence
    how many calls the nonce is valid for.
    
    Acked-by: Tom Lendacky <thomas.lendacky@amd.com>
    Signed-off-by: Mario Limonciello <mario.limonciello@amd.com>
    Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>

diff --git a/drivers/crypto/ccp/psp-dev.h b/drivers/crypto/ccp/psp-dev.h
index 505e4bdeaca8..8a4de69399c5 100644
--- a/drivers/crypto/ccp/psp-dev.h
+++ b/drivers/crypto/ccp/psp-dev.h
@@ -40,6 +40,7 @@ struct psp_device {
 	void *sev_data;
 	void *tee_data;
 	void *platform_access_data;
+	void *dbc_data;
 
 	unsigned int capability;
 };